dao2caom2/telescopes.py

```diff
@@ -318,46 +565,118 @@ def get_geo(self):
         result = None
         if obs_type in ['dark', 'object']:
             result = 0.0
         return result
 
-    def get_position_function_cd22(self, header):
-        obs_type = header.get('OBSTYPE')
+    def get_position_function_cd22(self, ext):
+        obs_type = self._headers[ext].get('OBSTYPE')
         result = None
         if obs_type in ['dark', 'object']:
             # DB - set entrance aperture to a fixed 5" by 5" because of lack
             # of detailed information
             result = 0.001388
         return result
 
-    def get_position_function_coord1_pix(self, header):
-        return self.get_position_function_coord2_pix(header)
+    def get_position_function_coord1_pix(self, ext):
+        return self.get_position_function_coord2_pix(ext)
 
-    def get_position_function_coord2_pix(self, header):
-        obs_type = header.get('OBSTYPE')
+    def get_position_function_coord2_pix(self, ext):
+        obs_type = self._headers[ext].get('OBSTYPE')
         result = None
         if obs_type in ['dark', 'object']:
             result = 1.0
         return result
 
-    def get_telescope_name(self):
+    def get_telescope_name(self, ign):
         return None
 
-    def get_time_axis_val(self, header):
-        return ac.get_datetime(header.get('DATE-OBS'))
+    def get_time_axis_val(self, ext):
+        return ac.get_datetime(self._headers[ext].get('DATE-OBS'))
+
+    def get_geo_x(self, ext):
+        x, ign1, ign2 = self.get_geo()
+        return x
+
+    def get_geo_y(self, ext):
+        x, y, ign2 = self.get_geo()
+        return y
+
+    def get_geo_z(self, ext):
+        x, ign1, z = self.get_geo()
+        return z
+
+    def get_obs_intent(self, ext):
+        obs_type = self._headers[ext].get('OBSTYPE')
+        if obs_type == 'object':
+            intent = ObservationIntentType.SCIENCE
+        else:
+            intent = ObservationIntentType.CALIBRATION
+        return intent
+
+    def get_time_exposure(self, ext):
+        exptime = mc.to_float(self._headers[ext].get('EXPTIME'))
+        ncombine = mc.to_float(self._headers[ext].get('NCOMBINE'))
+        if ncombine is not None:
+            # DB - approximation of exposure time for products (assume identical
+            # EXPTIME)
+            exptime *= ncombine
+        return exptime
+
+    def get_time_resolution(self, ext):
+        exptime = mc.to_float(self._headers[ext].get('EXPTIME'))
+        ncombine = mc.to_float(self._headers[ext].get('NCOMBINE'))
+        if ncombine is None:
+            ncombine = 1
+        else:
+            exptime = exptime * ncombine
+        return exptime / ncombine
+
+    def get_position_function_coord1_val(self, ext):
+        ra, ignore_dec = self._get_position(ext)
+        return ra
+
+    def get_position_function_coord2_val(self, ext):
+        ignore_ra, dec = self._get_position(ext)
+        return dec
+
+    def _get_position(self, ext):
+        from astropy.coordinates import SkyCoord, FK5
```

Comment:
Shouldn't these imports go at the top of the file?

Reply:
Whoops, yes. Thanks for catching that.